pkg/tasks/grafana.go

```diff
@@ -120,6 +122,11 @@ func (t *GrafanaTask) create(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling Grafana Datasources Secret failed")
 	}
 
+	err = t.client.DeleteSecretByNamespaceAndName(ctx, sds.GetNamespace(), grafanaLegacyDataSourceSecretName)
```

Comment:
Can we not reuse the DeleteSecret method by creating a Secret with name and namespace?

Reply:
Fixed it.

/hold for testing!